nixos/tests/systemd-initrd-luks-keyfile.nix

```diff
@@ -32,7 +32,7 @@ in {
         };
       };
       virtualisation.bootDevice = "/dev/mapper/cryptroot";
-      boot.initrd.systemd.contents."/etc/cryptroot.key".source = keyfile;
+      boot.initrd.secrets."/etc/cryptroot.key" = keyfile;
```

Comment:
It would be good to test this with the file not being loaded from the nix store, as one would want it in "real" use.

Reply:
I tried but I don't think this is really possible because the specialisation is built in a sandbox for adding the bootloader:
```
mounting Nix store...
mounting host's temporary directory...
starting stage 2 (/nix/store/nx7v7ywfmwpf3q124r4hrsagyhzk3ywp-vm-run-stage2)
Creating new GPT entries in memory.
Setting name!
partNum is 0
Setting name!
partNum is 1
The operation has completed successfully.
mkfs.fat 4.2 (2021-01-31)
Hidden (32) does not match sectors (63)
Initializing machine ID from random generator.
Created "/boot/EFI".
Created "/boot/EFI/systemd".
Created "/boot/EFI/BOOT".
Created "/boot/loader".
Created "/boot/loader/entries".
Created "/boot/EFI/Linux".
Copied "/nix/store/nbh44x952xhivydhgi5a7bgvnb6l35b1-systemd-250.4/lib/systemd/boot/efi/systemd-bootx64.efi" to "/boot/EFI/systemd/systemd-bootx64.efi".
Copied "/nix/store/nbh44x952xhivydhgi5a7bgvnb6l35b1-systemd-250.4/lib/systemd/boot/efi/systemd-bootx64.efi" to "/boot/EFI/BOOT/BOOTX64.EFI".
Created /etc/machine-info with KERNEL_INSTALL_LAYOUT=bls
Random seed file /boot/loader/random-seed successfully written (512 bytes).
cp: cannot stat '/keyfile': No such file or directory
Traceback (most recent call last):
  File "/nix/store/h0v4frq215x98xw97kdpnhzf6hm1l8h4-systemd-boot", line 316, in <module>
    main()
  File "/nix/store/h0v4frq215x98xw97kdpnhzf6hm1l8h4-systemd-boot", line 280, in main
    write_entry(*specialisation, machine_id)
  File "/nix/store/h0v4frq215x98xw97kdpnhzf6hm1l8h4-systemd-boot", line 113, in write_entry
    subprocess.check_call([append_initrd_secrets, "/boot%s" % (initrd)])
  File "/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/g4zkz489dzv975mmwasicap2186ilyas-append-initrd-secrets/bin/append-initrd-secrets', '/boot/efi/nixos/yqyxd44kq80ld7l9vk3nsn7qcrixpbdi-initrd-linux-5.15.36-initrd.efi']' returned non-zero
exit status 1.
[    3.014534] reboot: Power down
```

Reply:
That seems like a problem that should be addressed, but it seems reasonable enough to consider it out of scope for this PR :+1: